modules/charon-core/src/main/java/org/wso2/charon3/core/schema/ServerSideValidator.java

```diff
@@ -248,11 +252,56 @@ public static AbstractSCIMObject validateUpdatedSCIMObject(AbstractSCIMObject ol
             // Check for required attributes.
             validateSCIMObjectForRequiredAttributes(newObject, resourceSchema);
         }
+
+        Map<String, Attribute> attributes = validatedObject.getAttributeList();
+
+        for (Map.Entry<String, Attribute> entry : attributes.entrySet()) {
+            String key = entry.getKey();
+            Attribute value = entry.getValue();
+
+            if (value instanceof SimpleAttribute && StringUtils.equals(key, SCIMConstants.UserSchemaConstants.LOCALE)) {
+                String localeAttributeValue = ((SimpleAttribute) value).getValue().toString();
+
+                if (!isValidLocale(localeAttributeValue)) {
+                    throw new BadRequestException
+                            ("Provided locale value " + localeAttributeValue + " is invalid");
+                }
+            }
+
+        }
+
+
         // Check for schema list.
         validateSchemaList(validatedObject, resourceSchema);
         return validatedObject;
     }
 
+    public static boolean isValidLocale(String localeStr) {
```

Comment:
```suggestion
    private static boolean isValidLocale(String locale) {
```

Reply:
Fixed with [8a19b9f](https://github.com/wso2/charon/pull/415/commits/8a19b9ffd783b44b8cf29e8b3c0d9b3484570341) and [06c1403](https://github.com/wso2/charon/pull/415/commits/06c1403c1e9a4f13571667e751e8b14c515c5d77)